compiler/rustc_lint/src/builtin.rs

```diff
@@ -1424,8 +1424,24 @@ impl<'tcx> LateLintPass<'tcx> for UnreachablePub {
     }
 
     fn check_field_def(&mut self, cx: &LateContext<'_>, field: &hir::FieldDef<'_>) {
-        if matches!(cx.tcx.parent_hir_node(field.hir_id), Node::Variant(_)) {
-            return;
+        match cx.tcx.parent_hir_node(field.hir_id) {
+            Node::Variant(_) => return,
+            // We don't want to lint on struct/union fields whose parent def is
+            // not public. We therefore check if the parent is a struct/union and
```

Comment:
Need to tell why (e.g. https://github.com/rust-lang/rust/pull/126013#issuecomment-2152839205).